paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -179,6 +180,68 @@ class PaymentSheet internal constructor(
         IntentConfirmationInterceptor.createIntentCallback = createIntentCallback
     }
 
+    /**
+     * Builder to add optional callbacks to [PaymentSheet].
+     *
+     * @param resultCallback Called with the result of the payment after [PaymentSheet] is dismissed.
+     */
+    class Builder(internal val resultCallback: PaymentSheetResultCallback) {
+        internal var externalPaymentMethodConfirmHandler: ExternalPaymentMethodConfirmHandler? = null
+            private set
+        internal var createIntentCallback: CreateIntentCallback? = null
+            private set
+
+        /**
+         * @param handler Called when a user confirms payment for an external payment method.
+         */
+        fun externalPaymentMethodConfirmHandler(handler: ExternalPaymentMethodConfirmHandler) = apply {
+            externalPaymentMethodConfirmHandler = handler
+        }
+
+        /**
+         * @param intentCallback Called when the customer confirms the payment or setup.
+         */
+        fun createIntentCallback(intentCallback: CreateIntentCallback) = apply {
+            createIntentCallback = intentCallback
+        }
+
+        /**
+         * Returns a [PaymentSheet] and initializes callback handlers if respective callbacks are set
+         *
+         * @param activity The Activity that is presenting [PaymentSheet].
+         */
+        fun build(activity: ComponentActivity): PaymentSheet {
+            initializeCallbacks()
+            return PaymentSheet(DefaultPaymentSheetLauncher(activity, resultCallback))
+        }
+
+        /**
+         * Returns a [PaymentSheet] and initializes callback handlers if respective callbacks are set
+         *
+         * @param fragment the Fragment that is presenting the payment sheet.
+         */
+        fun build(fragment: Fragment): PaymentSheet {
+            initializeCallbacks()
+            return PaymentSheet(DefaultPaymentSheetLauncher(fragment, resultCallback))
+        }
+
+        /**
+         * Returns a [PaymentSheet] composable attached to the [Activity] that owns the calling context
+         * and initializes callback handlers if respective callbacks are set
+         */
+        @Composable
+        fun build(): PaymentSheet = rememberPaymentSheet(this)
```

Comment:
This feels a bit odd. Should we just have the `rememberPaymentSheet(builder: Builder)` method introduced below?